processors/baggage/baggagetrace/doc.go

```diff
@@ -2,4 +2,56 @@
 // SPDX-License-Identifier: Apache-2.0
 
 // Package baggagetrace implements a baggage span processor.
+//
+// This is an OpenTelemetry [Span Processor] that reads key/values stored in
+// [Baggage] in the starting span's parent context and adds them as attributes
+// to the span.
+//
+// Keys and values added to Baggage will appear on all subsequent child spans for
+// a trace within this service *and* will be propagated to external services via
+// propagation headers.
+// If the external services also have a Baggage span processor, the keys and
+// values will appear in those child spans as well.
+//
+// ⚠️ Waning ⚠️
+// To repeat: a consequence of adding data to Baggage is that the keys and values
+// will appear in all outgoing HTTP baggage headers from the application.
+//
+// Do not put sensitive information in Baggage.
+//
+// # Usage
+//
+// Add the span processor when configuring the tracer provider.
+//
+// To configure the span processor to copy all baggage entries during
+// configuration:
+//
+//	import (
+//		"go.opentelemetry.io/contrib/processors/baggage/baggagetrace"
+//	)
+//
+//	tp := trace.NewTracerProvider(
+//		trace.WithSpanProcessor(baggagetrace.New(baggagetrace.AllowAllBaggageKeys)),
+//		// ...
+//	)
+//
+// Alternatively, you can provide a custom baggage key predicate to select
+// which baggage keys you want to copy.
+//
+// For example, to only copy baggage entries that start with 'my-key':
+//
+//	baggagetrace.New(func(baggageKey string) bool {
+//		return strings.HasPrefix(baggageKey, "my-key")
+//	})
+//
+// For example, to only copy baggage entries that match the regex '^key.+':
+//
+//	expr := regexp.MustCompile(`^key.+`)
+//
+//	baggagetrace.New(func(baggageKey string) bool {
+//		return expr.MatchString(baggageKey)
+//	})
```

Comment:
From 

> Prefer using [Examples](https://pkg.go.dev/testing#hdr-Examples) instead of putting code snippets in Go doc comments. In some cases, you can even create [Testable Examples](https://go.dev/blog/examples).

Reply:
Added examples in [111bf75](https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5619/commits/111bf75ee014958330702c54cee0758418fe54b9)